src/server/nuclearnet/web_socket_proxy_nuclearnet_server.ts

```diff
@@ -93,6 +102,83 @@ class WebSocketServerClient {
   }
 
   private onPacket = (event: string, packet: NUClearNetPacket) => {
+    const peerKey = this.getPeer(this.processors.keys(), packet.peer)
+    if (peerKey) {
+      const processor = this.processors.get(peerKey)
+      if (processor) {
+        processor.onPacket(event, packet)
+      }
+    }
+  }
+
+  /**
+   * Look for the needle 'peer' inside the haystack 'peers'. They might not be the same object reference.
+   */
+  private getPeer(peers: Iterable<NUClearNetPeer>, peer: NUClearNetPeer): NUClearNetPeer | undefined {
+    return Array.from(peers).find(otherPeer => {
+      return otherPeer.name === peer.name && otherPeer.address === peer.address && otherPeer.port === peer.port
+    })
+  }
+}
+
+class RobotMessageProcessor {
+  private eventQueueSize: Map<string, number>
+
+  // The maximum number of packets of a unique type to send before receiving acknowledgements.
+  private limit: number
```

Comment:
I wonder if you can use tcp's slow start scheme here to find the limit dynamically. Although the advantage to that would be latency masking rather than increase in bandwidth. Could be useful for situations like the heroku app.